commons/src/main/java/org/acme/common/ManagedHttpClientFactory.java

```diff
@@ -65,7 +67,11 @@
 import java.util.Arrays;
 import java.util.Map;
 
-public final class ManagedHttpClientFactory {
+@ApplicationScoped
+public class ManagedHttpClientFactory {
```

Comment:
Looking at this code, I think we can drop both `ManagedHttpClientFactory` and `ManagedHttpClient`, and replace it with an `@ApplicationScoped` `HttpClient` instead. The logic of `newManagedHttpClient` can be moved to a `@Produces` method. 

It looks like `ManagedHttpClientFactory` is more or less a workaround due to the lack of an Inversion of Control container like [ArC](https://quarkus.io/guides/cdi-reference) in vanilla DT. Instances of HttpClient are thread safe by design, so there's no need to create more than one.

Reply:
I have adopted this change :)